third_party/move/move-compiler/src/expansion/ast.rs

```diff
@@ -377,6 +380,19 @@ pub enum ModuleAccess_ {
     // ModuleAccess(module_ident, member_ident, optional_variant_ident)
     ModuleAccess(ModuleIdent, Name, Option<Name>),
 }
+
+impl ModuleAccess_ {
+    fn get_name(&self) -> Name {
+        match self {
+            ModuleAccess_::Name(n) | ModuleAccess_::ModuleAccess(_, n, _) => *n,
+        }
+    }
+
+    pub fn is_struct_or_schema_name(&self) -> bool {
```

Comment:
I would keep the function name same as the original here, despite being wordy.  It's more precise, and `is_valid...` implies that the form is ok but it might not be a struct or schema.